src/domain/parties/parties.js

```diff
@@ -77,37 +83,43 @@ const getPartiesByTypeAndID = async (req) => {
  *
  * @description This sends a callback to inform participant of successful lookup
  *
- * @param {object} req The request object from the Hapi server
+ * @param {object} headers - incoming http request headers
+ * @param {object} params - uri parameters of the http request
+ * @param {string} method - http request method
+ * @param {object} payload - payload of the request being sent out
+ * @param {string} dataUri - encoded payload of the request being sent out
  */
-const putPartiesByTypeAndID = async (req) => {
+const putPartiesByTypeAndID = async (headers, params, method, payload, dataUri) => {
   try {
     Logger.info('parties::putPartiesByTypeAndID::begin')
-    const requesterParticipant = await participant.validateParticipant(req.headers['fspiop-source'])
-    const type = req.params.Type
-    if (Object.values(Enums.type).includes(type)) {
+    const requesterParticipant = await participant.validateParticipant(headers[Enums.Http.Headers.FSPIOP.SOURCE])
+    const type = params.Type
+    if (Object.values(Enums.Accounts.PartyAccountTypes).includes(type)) {
       if (requesterParticipant) {
-        const destinationParticipant = await participant.validateParticipant(req.headers['fspiop-destination'])
+        const destinationParticipant = await participant.validateParticipant(headers[Enums.Http.Headers.FSPIOP.DESTINATION])
         if (destinationParticipant) {
           const options = {
             partyIdType: type,
-            partyIdentifier: req.params.ID
+            partyIdentifier: params.ID
           }
-          await participant.sendRequest(req, destinationParticipant.data.name, Enums.endpointTypes.FSPIOP_CALLBACK_URL_PARTIES_PUT, Enums.restMethods.PUT, req.payload, options)
+          const decodedPayload = decodePayload(dataUri, { asParsed: false })
+          await participant.sendRequest(headers, destinationParticipant.data.name, Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTIES_PUT, Enums.Http.RestMethods.PUT, decodedPayload.body.toString(), options)
           Logger.info('parties::putPartiesByTypeAndID::end')
         } else {
-          await participant.sendErrorToParticipant(req, requesterParticipant, Enums.endpointTypes.FSPIOP_CALLBACK_URL_PARTIES_PUT_ERROR,
-            util.buildErrorObject(Errors.ErrorObject.DESTINATION_FSP_NOT_FOUND_ERROR, [{key: type, value: req.params.ID}]))
+          await participant.sendErrorToParticipant(headers[Enums.Http.Headers.FSPIOP.SOURCE], Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTIES_PUT_ERROR,
+            ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.DESTINATION_FSP_ERROR).toApiErrorObject(), headers, params)
         }
       } else {
         Logger.error('Requester FSP not found')
-        // TODO: handle issue where requester fsp not found
+        throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found').toApiErrorObject()
```

Comment:
`throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found').toApiErrorObject()` --> `throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found')`